### ISSUE TITLE
gmmhmm cannot work 

### ISSUE THREAD

**Issue by LINYANWANG** [ohjay/hmm_activity_recognition#1]
If option 'm_type' was set as 'gmm', the training will not converge. While 'GaussianHMM' can work and the training can converge.

**Comment by ohjay** [comment]
Hmm, I'm not sure how to reproduce this. Are you running one of the provided config files?
With Python 2.7, `hmmlearn` 0.2.0, and the `owen.yaml` config file, my models converge.

**Comment by fbiying87** [comment]
> Hmm, I'm not sure how to reproduce this. Are you running one of the provided config files?
> With Python 2.7, `hmmlearn` 0.2.0, and the `owen.yaml` config file, my models converge.

Should it be run on Python 2.7? May I also use Python 3.6?
Thanks!

**Comment by ohjay** [comment]
**Edit:** the question above appears to have been deleted. For anyone wondering, the commenter was asking whether they could use Python 3.6 instead of Python 2.7.

I've only tried Python 2.7, so I guess that's the only version I can endorse. Maybe Python 3.6 forces you to use different versions of the modules and they don't behave the same in all regards.

**Comment by ohjay** [comment]
(If time permits, I'll test the code with Python 3.6 later this week and get back to you.)

**Comment by fbiying87** [comment]
> (If time permits, I'll test the code with Python 3.6 later this week and get back to you.)

Thanks for your reply. I tested it on Python 3.6. It seems to work for feature extraction and model building. But my current problem is, that some of the models like e.g. walking returns always nan as score. Therefore after sort, it always return 0 for this activity. Is it maybe possible to share your model with me? I want to reproduce the accuracy you got. It should be around 50% for each activity right?

Thanks!

**Comment by ohjay** [comment]
Sure, [here's a ZIP file with some pre-trained models](https://drive.google.com/file/d/10cEhDoKnUWxJNUKEZk14EAOe0OeMhpxO/view?usp=sharing). If you extract the contents of the ZIP file in the project's root directory, then you can evaluate the models' performance with

```
python main.py classify config/quickstart.yaml
```

Using these models, I observe classification accuracies ranging from 58% to 80% on my validation split (the one generated by `./get_data.sh`).

**Comment by ohjay** [comment]
Hey @fbiying87, I managed to run a couple of tests. Python 3.6 seems to work as long as you use the exact versions of the modules that are specified in [`requirements.txt`](https://github.com/ohjay/hmm_activity_recognition/blob/master/requirements.txt). It's when you switch those up that things get a little iffy. After I upgraded `hmmlearn` and `scikit-learn` to their latest versions, I started seeing NaN warnings, so there may be numerical issues somewhere.

For the time being, I recommend you just set up a virtual environment with the supported dependency versions. [`quickstart.sh`](https://github.com/ohjay/hmm_activity_recognition/blob/master/quickstart.sh) might help you get started with that.

**Comment by fbiying87** [comment]
Hi @ohjay , thanks a lot for your reply. I figured it would be something like this. I used the latest hmmlearn and opencv version. Some models were droped due to NaN predictions. That's why the counts of the models were different for each activity. I will try to use the equal version of the requirements.txt to reproduce the results. Thanks again.
